x-pack/plugins/observability_solution/observability_ai_assistant_app/server/functions/query/correct_query_with_actions.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { validateQuery, getActions } from '@kbn/esql-validation-autocomplete';
+import { getAstAndSyntaxErrors } from '@kbn/esql-ast';
+
+const fixedQueryByOneAction = async (queryString: string) => {
+  const { errors } = await validateQuery(queryString, getAstAndSyntaxErrors);
+
+  const actions = await getActions(queryString, errors, getAstAndSyntaxErrors, {
+    relaxOnMissingCallbacks: true,
+  });
+
+  if (actions.length) {
+    const [firstAction] = actions;
+    const range = firstAction.edits[0].range;
+    const correctText = firstAction.edits[0].text;
+    const problematicString = queryString.substring(range.startColumn - 1, range.endColumn - 1);
```

Comment:
are the columns not zero-indexed that this is necessary?

Reply:
Correct